static_report/src/hooks/useComparisonChart.ts

```diff
@@ -0,0 +1,150 @@
+import * as d3 from 'd3';
+import { useEffect, RefObject } from 'react';
+
+import { getChartTooltip, formatNumber } from '../utils';
+
+const GROUPED = ['base', 'input'];
+const X_PADDING = 0.5;
+const X_SUB_PADDING = 0.05;
+const TOOLTIPS_BG_COLOR = 'var(--chakra-colors-gray-500)';
+const BASE_CHART_COLOR = 'var(--chakra-colors-blue-100)';
+const INPUT_CHART_COLOR = 'var(--chakra-colors-blue-300)';
+
+export function useComparisonChart<T extends SVGSVGElement>({
+  target,
+  data,
+  dimensions,
+}: {
+  target: RefObject<T>;
+  // FIXME: data types
+  data: any;
+  dimensions: DOMRect;
+}) {
+  useEffect(() => {
+    if (!target || !dimensions || !data) {
+      return;
+    }
+
+    const svg = d3.select(target.current);
+
+    // X-Axis
+    const xScale = d3
+      .scaleBand()
+      .domain(data.map((d) => d.label))
+      .range([0, dimensions.width])
+      .padding(X_PADDING);
+
+    const xSubGroup = d3
+      .scaleBand()
+      .domain(GROUPED)
+      .range([0, xScale.bandwidth()])
+      .padding(X_SUB_PADDING);
+
+    const xAxis = d3.axisBottom(xScale).tickFormat((value, i) => {
+      const xAxisItemLength = xScale.domain().length - 1;
+
+      if (i === 0 || i === xAxisItemLength / 2 || i === xAxisItemLength) {
```

Comment:
For the middle ticker, should we handle both odd/even ticker lengths?
![image](https://user-images.githubusercontent.com/11186480/176371413-7f36faba-9e76-41e3-aca1-3a50fc3d76c2.png)

Reply:
Clarify with Jon, resolved.